lib/parsers/open-api-3.1.js

```diff
@@ -32,6 +26,26 @@ class JoiOpenApiSchemaParser extends JoiJsonSchemaParser {
   _getLocalSchemaBasePath() {
     return '#/components/schemas'
   }
+
+  _copyMetasToSchema(joiSpec, schema) {
+    if (!_.isEmpty(joiSpec.metas)) {
+      _.each(joiSpec.metas, meta => {
+        _.each(meta, (value, key) => {
+          if (key.startsWith('x-') || key === 'deprecated' || key === 'readOnly' || key === 'writeOnly') {
+            schema[key] = value
+          }
+        })
+      })
+    }
+  }
+
+  _moveSchemaToDefinitions(schema, joiSpec, definitions, schemaId) {
+    this._copyMetasToSchema(joiSpec, schema)
+    definitions[schemaId] = schema
+    return {
+      $ref: `${this._getLocalSchemaBasePath()}/${schemaId}`
+    }
+  }
```

Comment:
I wonder if `_copyMetasToSchema` and `_moveSchemaToDefinitions` can be placed at `parsers/json.js` to standardize behavior.  Then we can have an overridable function `_isMetaQualifiedToCopy` or something  similar, then each parser can return true/false depending on the key checking logic, and then it do the copy action accordingly.